openshift/3scale_backend.conf

```diff
@@ -22,14 +22,34 @@ ThreeScale::Backend.configure do |config|
   config.internal_api.user = "#{ENV['CONFIG_INTERNAL_API_USER']}"
   config.internal_api.password = "#{ENV['CONFIG_INTERNAL_API_PASSWORD']}"
   config.queues.master_name = "#{ENV['CONFIG_QUEUES_MASTER_NAME']}"
+  config.queues.username = "#{ENV['CONFIG_QUEUES_USERNAME']}"
+  config.queues.password = "#{ENV['CONFIG_QUEUES_PASSWORD']}"
+  config.queues.ssl =  parse_boolean_env('CONFIG_QUEUES_SSL')
+  config.queues.ssl_params = {
+    ca_file: "#{ENV['CONFIG_QUEUES_CA_FILE']}",
+    cert: "#{ENV['CONFIG_QUEUES_CERT']}",
+    key: "#{ENV['CONFIG_QUEUES_PRIVATE_KEY']}"
+  }
   config.queues.sentinels = "#{ENV['CONFIG_QUEUES_SENTINEL_HOSTS'] && !ENV['CONFIG_QUEUES_SENTINEL_HOSTS'].empty? ? ENV['CONFIG_QUEUES_SENTINEL_HOSTS'] : ENV['SENTINEL_HOSTS']}"
+  config.queues.sentinel_username = "#{ENV['CONFIG_QUEUES_SENTINEL_USERNAME']}"
+  config.queues.sentinel_password = "#{ENV['CONFIG_QUEUES_SENTINEL_PASSWORD']}"
   config.queues.role = "#{ENV['CONFIG_QUEUES_SENTINEL_ROLE']}".to_sym
   config.queues.connect_timeout = parse_int_env('CONFIG_QUEUES_CONNECT_TIMEOUT')
   config.queues.read_timeout = parse_int_env('CONFIG_QUEUES_READ_TIMEOUT')
   config.queues.write_timeout = parse_int_env('CONFIG_QUEUES_WRITE_TIMEOUT')
   config.queues.max_connections = parse_int_env('CONFIG_QUEUES_MAX_CONNS')
   config.redis.proxy = "#{ENV['CONFIG_REDIS_PROXY']}"
+  config.redis.username = "#{ENV['CONFIG_REDIS_USERNAME']}"
+  config.redis.password = "#{ENV['CONFIG_REDIS_PASSWORD']}"
+  config.redis.ssl =  parse_boolean_env('CONFIG_REDIS_SSL')
```

Comment:
Do we have a use case where user can't just set `rediss://` instead? Even for centinels I understand we set uri. So we can infer this from the uri. I mean, can we remove this option?

Reply:
Initially, I thought it could be removed, I even created a PR for porta: https://github.com/3scale/porta/pull/3808. However, there's an scenario where it's required: when using sentinels, and providing the name through the `:name` param. In that scenario the `:url` is not needed so there's no other way to set `ssl: true` rather than the `:ssl` param.

For the porta PR what I decided is to take the ssl value from the URL but only in case it's not provided as a parameter. I think in general is good to have the two ways to set the values: first by param, if not present, then from the URL. That's how it works for `:name` and `:db`.

What I could do is to implement the same behavior for `:ssl` as I did in that porta PR.

Reply:
Duplicating the configuration is a double-edge sword as we need to resolve situations of conflict.

For example, with the above implementation, if user has configured `ssl` to `false` but used `rediss` schema, then schema will take precedence. If `ssl` is set to `true` but schema is `redis`, then `ssl` will take precedence.

I'm not saying this behavior is bad. Actually it is a safe behavior because TLS connection type will take precedence if it is configures on either place. It should be well documented though, to avoid confusion.

I still would personally opt to require `url` to be configured in case TLS connection is desired and avoid double config. But if you prefer having both - that's also fine with me, with configuration option descriptions being made more thorough.

Reply:
I prefer to have both. I'll document the precedence.